### ISSUE TITLE
DNS over HTTPS

### ISSUE THREAD

**Issue by sadreck** [ctxis/SnitchDNS#5]
Not sure if this is actually needed, but it'd be a good feature to add.

**Comment by johnneerdael** [comment]
Definitely would be great for showing DNS tunneling over DNS over HTTPS

**Comment by rwjack** [comment]
This can be achieved even easier by allowing custom ports in the DNS forwarding section.
![image](https://user-images.githubusercontent.com/59068073/155751104-779981f1-2546-4e5c-8b43-20f32e09496c.png)
Eg. I could just forward to my local cloudflared instance on port 5053 which does the DOH.

**Comment by sadreck** [comment]
I don't believe that can be done because the Python library I'm using ([Twisted Matrix](https://twistedmatrix.com/trac/)) doesn't support DNS over HTTPS and will assume that it's forwarding the query to a standard DNS server.

Unless you mean that the service running on the target port does the conversion from DNS to DoH?

**Comment by rwjack** [comment]
> Unless you mean that the service running on the target port does the conversion from DNS to DoH?

That is exactly what I mean. Twisted should just be able to forward a plain DNS request to a specified IP:PORT mapping, instead of just an IP mapping.

This is the image I wanted to paste in my upper comment:
![image](https://user-images.githubusercontent.com/59068073/155758120-8f23fadf-6b6f-43d3-a688-dbd5c543f712.png)

**Comment by sadreck** [comment]
I've implemented support for `IP:Port`, but I'll keep this ticket open for when I have time to implement actual DoH
